test/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_service_test.clj

```diff
@@ -385,3 +385,22 @@
        (is (logged?
             #"shutdown-on-error triggered because of exception"
             :error))))))
+
+(deftest ^:integration overflow-test
+  (let  [watch-dir  (fs/temp-dir "overflow")
+        results  (atom  [])
+        callback  (fn  [events]  (swap! results concat events))]
+    (with-app-with-config
+     app watch-service-and-deps  {}
+     (let  [service  (tk-app/get-service app :FilesystemWatchService)]
+       (watch! service watch-dir callback))
+     ;; From experimenting the buffer seems to fill at 512 events.
+     ;; However the number of events varies by operating system.
+     (let  [files  (for  [n  (range 1000)]
```

Comment:
That's so weird. I copy pastad your [reproducer](https://github.com/KevinCorcoran/trapperkeeper-filesystem-watcher/commit/53f66ce4df19a47c1c51da30d31899cdf4c934e0) to start this ticket but when I pasted my editor inserted extra spaces everywhere in the test.... I've addressed this in an incoming commit. Apologies.